Riot/Modules/Home/VersionCheck/VersionCheckAlertViewController.swift

```diff
@@ -0,0 +1,110 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import Reusable
+
+protocol VersionCheckAlertViewControllerDelegate: AnyObject {
+    func alertViewControllerDidRequestDismissal(_ alertViewController: VersionCheckAlertViewController)
+    func alertViewControllerDidRequestAction(_ alertViewController: VersionCheckAlertViewController)
+}
+
+struct VersionCheckAlertViewControllerDetails {
+    let title: String
+    let subtitle: String
+    let actionButtonTitle: String
+}
+
+class VersionCheckAlertViewController: UIViewController {
+    
+    @IBOutlet private var alertContainerView: UIView!
+    @IBOutlet private var titleLabel: UILabel!
+    @IBOutlet private var subtitleLabel: UILabel!
+    
+    @IBOutlet private var dismissButton: UIButton!
+    @IBOutlet private var actionButton: UIButton!
+    
+    private var themeService: ThemeService!
+    private var details: VersionCheckAlertViewControllerDetails?
+    
+    weak var delegate: VersionCheckAlertViewControllerDelegate?
+    
+    static func instantiate(themeService: ThemeService) -> VersionCheckAlertViewController {
+        let versionCheckAlertViewController = VersionCheckAlertViewController(nibName: nil, bundle: nil)
+        versionCheckAlertViewController.themeService = themeService
+        versionCheckAlertViewController.modalPresentationStyle = .overFullScreen
+        versionCheckAlertViewController.modalTransitionStyle = .crossDissolve
+        return versionCheckAlertViewController
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        actionButton.layer.masksToBounds = true
+        actionButton.layer.cornerRadius = 6.0
+        
+        configureWithDetails(details)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(updateTheme), name: .themeServiceDidChangeTheme, object: nil)
+        updateTheme()
+    }
+    
+    func configureWithDetails(_ details: VersionCheckAlertViewControllerDetails?) {
+        guard let details = details else {
+            return
+        }
+        
+        guard self.isViewLoaded else {
+            self.details = details
+            return
+        }
+        
+        titleLabel.text = details.title
+        actionButton.setTitle(details.actionButtonTitle, for: .normal)
+        
+        let attributedSubtitle = NSMutableAttributedString(string: details.subtitle)
+        
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineHeightMultiple = 1.2
```

Comment:
Is this constant only applying to this banner? Otherwise we should think about having it in the DesignKit

Reply:
In this particular case it's not, Galle was explicit about it only being a thing here 👍